akka-stream-tests/src/test/scala/akka/stream/FusingSpec.scala

```diff
@@ -79,12 +90,23 @@ class FusingSpec extends StreamSpec {
       verify(fused, modules = 1, downstreams = 6)
     }
 
+    "properly fuse an async FusedGraph that has been extended (no AsyncBoundary)" in {
+      val src = Future(Fusing.aggressive(graph(false)))
+      val fused = Fusing.aggressive(Source.fromFutureGraph(src).to(Sink.head))
+      asyncVerify(fused, modules = 1, downstreams = 1)
+    }
+
     "properly fuse a FusedGraph that has been extended (with AsyncBoundary)" in {
       val src = Fusing.aggressive(graph(true))
       val fused = Fusing.aggressive(Source.fromGraph(src).to(Sink.head))
       verify(fused, modules = 2, downstreams = 6)
     }
 
+    "properly fuse an async FusedGraph that has been extended (with AsyncBoundary)" in {
+      val src = Future(Fusing.aggressive(graph(true)))
+      val fused = Fusing.aggressive(Source.fromFutureGraph(src).to(Sink.head))
+      asyncVerify(fused, modules = 1, downstreams = 1)
+    }
```

Comment:
I don't think these tests belong here. This Spec is solely used for fusing related issues, this feature does not introduce any new things to fusing, so these tests seems like at the wrong place. Probably SourceFromFutureGraphSpec is the place where these should go.